### PR TITLE
Add reference field - Closes #813

### LINKED ISSUE #813: Add support for reference field in transactions
### Expected behaviour
Lisk core 1.0.0 will add support for optional data field https://github.com/LiskHQ/lisk/issues/26
Lisk Hub should have a field in the send module to support it and display it in transactions list.

Design: https://taikonauten.invisionapp.com/d/main/#/console/12665756/284010013/preview

**should also be part of the launch protocol.**

### Actual behaviour
There is no support for reference field.

### PR DESCRIPTION
**Pull request by michaeltomasik**
### What was the problem?
No reference field in Transferbox
### How did I fix it?
Added Reference field
### How to test it?
Go to Wallet and make transaction with reference field and check if transaction have referance on transactionDetailView
### Review checklist
- The PR solves #813
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/components/sendWritable/send.js [hunk 1]
```diff
@@ -37,24 +41,30 @@ class SendWritable extends React.Component {
...
   }
 
-  validateInput(name, value) {
-    if (!value) {
+  validateInput(name, value, required) {
+    const byteCount = encodeURI(value).split(/%..|./).length - 1;
+    if (!value && required) {
       return this.props.t('Required');
+    } else if (name === 'reference' && byteCount > 64) {
+      return this.props.t('Max 64 bytes message');
```
Comment (slaweet) [inline src/components/sendWritable/send.js]: I would say `Maximum length exceeded` in order not to confuse users with what bytes means

File: src/utils/api/transactions.js [hunk 2]
```diff
@@ -30,10 +30,12 @@ export const getTransactions = ({
 export const getSingleTransaction = ({ activePeer, id }) => activePeer.transactions.get({ id });
 
 export const unconfirmedTransactions = (activePeer, address, limit = 20, offset = 0, sort = 'timestamp:desc') =>
-  activePeer.node.getTransactions('unconfirmed', {
-    senderId: address,
-    recipientId: address,
-    limit,
-    offset,
-    sort,
+  new Promise(() => {
```
Comment (slaweet) [inline src/utils/api/transactions.js]: Why are you wrapping it with `new Promise`? `activePeer.node.getTransactions` already returns a promise. 
The pending transactions stay in the list, but not sure if it's related to this.

Comment (michaeltomasik) [reply src/utils/api/transactions.js]: I had a problem in one spot not sure now which one, I would have to investigate it and let you know maybe it is already fixed with a current changes.

Comment (slaweet) [reply src/utils/api/transactions.js]: Wrapping the promise here means that it never gets resolved, so you'll prevent if there was some error, but some feature will be broken.

**Pull request merged**